### ISSUE TITLE
[no-magic-numbers] Computed constant values

### ISSUE THREAD

**Issue by mustafa-cosar** [eslint/eslint#12535]
**What rule do you want to change?**
no-magic-numbers

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior

**Please provide some example code that this change will affect:**

```js
/* eslint no-magic-numbers: ["error", {
                "ignore": [-1, 0, 1, 2],
            }] */

const  MAX_IMAGE_SIZE_IN_BYTES = 10 * 1024 * 1024;
```
[ESLint demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLW1hZ2ljLW51bWJlcnM6IFtcImVycm9yXCIsIHtcbiAgICAgICAgICAgICAgICBcImlnbm9yZVwiOiBbLTEsIDAsIDEsIDJdLFxuICAgICAgICAgICAgfV0gKi9cblxuY29uc3QgIE1BWF9JTUFHRV9TSVpFX0lOX0JZVEVTID0gMTAgKiAxMDI0ICogMTAyNDsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEwLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6eyJjb21tb25qcyI6ZmFsc2UsIm5vZGUiOmZhbHNlfX19)

**What does the rule currently do for this code?**
It produces magic number errors for 10, 1024, 1024

**What will the rule do after it's changed?**
I will not produce any errors

**Are you willing to submit a pull request to implement this change?**
No, it is not blocking but I think this should be the default.

Also, I'm using the typescript-eslint extension of this rule. I opened an issue there but they said this issue belongs here.

https://github.com/typescript-eslint/typescript-eslint/issues/1171

**Comment by kaicataldo** [comment]
I’m 👍🏼 for this proposal, but I think it should be put behind an option and not be the default behavior. This will allow control over how strict users would like the rule to be.

**Comment by ljharb** [comment]
it’d be great to reopen this one
